test/java/blockchains/iaas/uni/stuttgart/de/adaptation/adapters/ethereum/TestEthereumAdapter.java

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Institute for the Architecture of Application System - University of Stuttgart
+ * Author: Akshay Patel
+ *
+ * This program and the accompanying materials are made available under the
+ * terms the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+package blockchains.iaas.uni.stuttgart.de.adaptation.adapters.ethereum;
+
+import blockchains.iaas.uni.stuttgart.de.adaptation.AdapterManager;
+import blockchains.iaas.uni.stuttgart.de.api.IAdapterExtension;
+import blockchains.iaas.uni.stuttgart.de.api.interfaces.BlockchainAdapter;
+import blockchains.iaas.uni.stuttgart.de.api.model.LinearChainTransaction;
+import blockchains.iaas.uni.stuttgart.de.connectionprofiles.ConnectionProfilesManager;
+import blockchains.iaas.uni.stuttgart.de.management.BlockchainPluginManager;
+import org.junit.jupiter.api.*;
+import org.pf4j.PluginState;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.crypto.CipherException;
+import org.web3j.crypto.ECKeyPair;
+import org.web3j.crypto.WalletUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+
+public class TestEthereumAdapter {
+    // Add -Dpf4j.pluginsDir=<plugin_storage_path> -DethereumPluginJarPath=<path_to_ethereum_plugin_jar>
+
+    private static final Logger log = LoggerFactory.getLogger(TestEthereumAdapter.class);
+    private static final String NETWORK_NAME = "eth-0";
+    private static final double REQUIRED_CONFIDENCE = 0.6;
+    private final BlockchainPluginManager pluginManager = BlockchainPluginManager.getInstance();
+    private final Path pluginPath = Paths.get(System.getProperty("ethereumPluginJarPath"));
+
+    @BeforeEach
+    public void setUp() {
+        clearPluginDirectory();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        clearPluginDirectory();
+    }
+
+    @Test
+    public void testLoadEthereumPlugin() throws IOException {
```

Comment:
I think we should only test the plugin management logic here without creating one test class per plugin, maybe create a parametrized test that accepts paths to some plugin jars along with connection profiles that are suitable for them. If you want to test if the loading worked fine, you try the `testConnection()` method of the adapter interface...
Unit testing of the individual adapters must strictly remain in their corresponding repositories.